File: proj3-coeasao/q11/depois/Usuario.ts
import { Comentario } from "./Comentario";
import { IConteudo } from "./IConteudo";
import { Post } from "./Post";

class Usuario {
    private conteudos: IConteudo[] = [];

    criarConteudo(conteudo: IConteudo): void {
        this.conteudos.push(conteudo);
    }

    getConteudos(): Array<IConteudo>{
        return this.conteudos;
    }
}


const adelia: Usuario = new Usuario()

const post: IConteudo = new Post("Publicacao")
const comentario: IConteudo = new Comentario("Publicacao")

adelia.criarConteudo(post)
adelia.criarConteudo(comentario)


console.log(adelia.getConteudos());

File: proj3-coeasao/q11/antes/usuario.java
package antes;

public class usuario {
    private List<Post> posts;
    private List<Comentario> comentarios;

    public void criarPost(String conteudo) {
        Post novoPost = new Post(conteudo);
        posts.add(novoPost);
    }

    public void criarComentario(String conteudo) {
        Comentario novoComentario = new Comentario(conteudo);
        comentarios.add(novoComentario);

    }

}
